t/unit/backends/test_database.py

```diff
@@ -33,11 +35,19 @@ def test_backend__pickle_serialization(self):
 
         tid2 = uuid()
         result = {'foo': 'baz', 'bar': SomeClass(12345)}
-        self.b.mark_as_done(tid2, result)
-        # is serialized properly.
-        rindb = self.b.get_result(tid2)
-        assert rindb.get('foo') == 'baz'
-        assert rindb.get('bar').data == 12345
+        request = MagicMock()
+        request.task = 'my_task'
+        request.args = ['a', 1]
+        request.kwargs = {'c': 3, 'd': 'e'}
```

Comment:
Are these the only types of inputs that can be received?  What about other things?  

You really need to look at all the different things that the kombu dumps function (which is what actually handles...  

https://github.com/celery/kombu/blob/master/kombu/serialization.py#L161

Reply:
I think that `args` is always a list and `kwargs` always a dict.
So, you mean something like:
```
request.args = ['a', 1, ['2', 3, {'b': 'x'}], {'c': 'd', 'e': 3, 'f': [5, 6, 7]}]
```
?

Reply:
Yes, but an array/dict containing what?

Reply:
The `content_encoding` that returns the `encode_content` function only depends on `self.serializer`. Which means that it will be the same for `result`, `meta`, `task_args`, `task_kwargs` and for all others objects serialized using the `encode_content` of the same DatabaseBackend instance.

Moreover, we're also ignoring the `content_encoding` of `meta` and there is error when we deserialize it using the same `decode_content` function.

Reply:
I don't get you.
Is it okay or not this example: `request.args = ['a', 1, ['2', 3, {'b': 'x'}], {'c': 'd', 'e': 3, 'f': [5, 6, 7]}]` ?
Let me know which other data types I could include.

Reply:
What I mean, is what if I pass an object as an arg?  What about somethign that's not json serializable like django's lazy gettext objects?  What about a file handle?  A string that's 1 million bytes?  etc.  Your tests handle the easy test cases, but those are the kinds of tests that aren't particularly helpful... try to break your own code ;)

Reply:
I included the custom testing class `SomeClass` in args and kwargs to make sure that it is able to serialize non standard objects.

Reply:
awesome thanks!